extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectionBase.java

```diff
@@ -54,14 +58,17 @@ public String pathParam(String name) {
     }
 
     public Uni<Void> sendText(String message) {
-        Uni<Void> uni = Uni.createFrom().completionStage(() -> webSocket().writeTextMessage(message).toCompletionStage());
+        Uni<Void> uni = Uni.createFrom()
+                .completionStage(() -> webSocket().writeTextMessage(message).toCompletionStage())
+                .invoke(() -> sendingInterceptor.onSend(message));
         return trafficLogger == null ? uni : uni.invoke(() -> {
             trafficLogger.textMessageSent(this, message);
         });
     }
 
     public Uni<Void> sendBinary(Buffer message) {
-        Uni<Void> uni = Uni.createFrom().completionStage(() -> webSocket().writeBinaryMessage(message).toCompletionStage());
+        Uni<Void> uni = Uni.createFrom().completionStage(() -> webSocket().writeBinaryMessage(message).toCompletionStage())
+                .invoke(sendingInterceptor.onSend(message));
```

Comment:
The same as in https://github.com/quarkusio/quarkus/pull/41956/files#r1688148640